fabric-armor-api-v1/src/main/java/net/fabricmc/fabric/mixin/armor/client/ArmorFeatureRendererMixin.java

```diff
@@ -0,0 +1,102 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.mixin.armor.client;
+
+import java.util.Map;
+
+import org.spongepowered.asm.mixin.Final;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.Unique;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+import net.minecraft.client.render.VertexConsumerProvider;
+import net.minecraft.client.render.entity.feature.ArmorFeatureRenderer;
+import net.minecraft.client.render.entity.feature.FeatureRenderer;
+import net.minecraft.client.render.entity.feature.FeatureRendererContext;
+import net.minecraft.client.render.entity.model.BipedEntityModel;
+import net.minecraft.client.util.math.MatrixStack;
+import net.minecraft.entity.EquipmentSlot;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.item.ArmorItem;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.Identifier;
+
+import net.fabricmc.api.EnvType;
+import net.fabricmc.api.Environment;
+import net.fabricmc.fabric.api.client.armor.v1.CustomModeledArmor;
+import net.fabricmc.fabric.api.client.armor.v1.CustomTexturedArmor;
+
+@Mixin(ArmorFeatureRenderer.class)
+@Environment(EnvType.CLIENT)
+public abstract class ArmorFeatureRendererMixin extends FeatureRenderer {
+	@Shadow
+	@Final
+	private static Map<String, Identifier> ARMOR_TEXTURE_CACHE;
+
+	public ArmorFeatureRendererMixin(FeatureRendererContext context) {
+		super(context);
+	}
+
+	@Unique
+	private LivingEntity storedEntity;
+	private EquipmentSlot storedSlot;
+
+	@Inject(method = "render", at = @At("HEAD"))
+	private void storeEntity(MatrixStack matrixStack, VertexConsumerProvider vertexConsumerProvider, int i, LivingEntity livingEntity, float f, float g, float h, float j, float k, float l, CallbackInfo ci) {
+		this.storedEntity = livingEntity;
+	}
+
+	@Inject(method = "renderArmor", at = @At("HEAD"))
+	private void storeSlot(MatrixStack matrices, VertexConsumerProvider vertexConsumers, LivingEntity livingEntity, EquipmentSlot slot, int i, BipedEntityModel bipedEntityModel, CallbackInfo ci) {
+		this.storedSlot = slot;
+	}
+
+	@Inject(method = "render", at = @At("RETURN"))
+	private void removeStored(MatrixStack matrixStack, VertexConsumerProvider vertexConsumerProvider, int i, LivingEntity livingEntity, float f, float g, float h, float j, float k, float l, CallbackInfo ci) {
+		this.storedEntity = null;
+		this.storedSlot = null;
+	}
+
+	@Inject(method = "getArmor", at = @At("RETURN"), cancellable = true)
+	private void getArmor(EquipmentSlot slot, CallbackInfoReturnable<BipedEntityModel<LivingEntity>> cir) {
```

Comment:
Injection method doesn't describe very well what this does. Maybe `selectArmorModel`